lib/rules/no-control-regex.js

```diff
@@ -83,8 +96,10 @@ module.exports = {
                 if (regex) {
                     const computedValue = regex.toString();
 
-                    if (hasControlCharacters(computedValue)) {
-                        context.report(node, "Unexpected control character in regular expression.");
+                    const controlCharacters = getControlCharacters(computedValue);
+
+                    if (controlCharacters.length > 0) {
+                        context.report(node, "Unexpected control character(s) in regular expression: " + controlCharacters.join(", ") + ".");
```

Comment:
Any chance this could use new-style reporting?

```js
context.report({
    node,
    message: "Unexpected control character(s) in regular expression: {{controlChars}}.",
    data: {
        controlChars: controlCharacters.join(", ")
    }
});
```